File: carddemo-java/src/main/java/com/aws/carddemo/model/UserSecurity.java
package com.aws.carddemo.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * User Security entity - migrated from COBOL copybook CSUSR01Y.cpy
 * Original COBOL record length: 80 bytes
 * Used for user authentication and authorization
 */
@Entity
@Table(name = "USRSEC")
public class UserSecurity {

    @Id
    @Column(name = "SEC_USR_ID", length = 8)
    private String secUsrId;

    @Column(name = "SEC_USR_FNAME", length = 20)
    private String secUsrFname;

    @Column(name = "SEC_USR_LNAME", length = 20)
    private String secUsrLname;

    @Column(name = "SEC_USR_PWD", length = 8)
    private String secUsrPwd;

    @Column(name = "SEC_USR_TYPE", length = 1)
    private String secUsrType;

    public UserSecurity() {
    }

    public UserSecurity(String secUsrId) {
        this.secUsrId = secUsrId;
    }

    public String getSecUsrId() {
        return secUsrId;
    }

    public void setSecUsrId(String secUsrId) {
        this.secUsrId = secUsrId;
    }

    public String getSecUsrFname() {
        return secUsrFname;
    }

    public void setSecUsrFname(String secUsrFname) {
        this.secUsrFname = secUsrFname;
    }

    public String getSecUsrLname() {
        return secUsrLname;
    }

    public void setSecUsrLname(String secUsrLname) {
        this.secUsrLname = secUsrLname;
    }

    public String getSecUsrPwd() {
        return secUsrPwd;
    }

    public void setSecUsrPwd(String secUsrPwd) {
        this.secUsrPwd = secUsrPwd;
    }

    public String getSecUsrType() {
        return secUsrType;
    }

    public void setSecUsrType(String secUsrType) {
        this.secUsrType = secUsrType;
    }

    public boolean isAdmin() {
        return "A".equals(secUsrType);
    }

    public boolean isUser() {
        return "U".equals(secUsrType);
    }

    @Override
    public String toString() {
        return "UserSecurity{" +
                "secUsrId='" + secUsrId + '\'' +
                ", secUsrFname='" + secUsrFname + '\'' +
                ", secUsrLname='" + secUsrLname + '\'' +
                ", secUsrType='" + secUsrType + '\'' +
                '}';
    }
}

File: carddemo-batch-python/src/carddemo_batch/models/user_security.py
"""UserSecurity model - migrated from COBOL copybook CSUSR01Y.cpy (80 bytes)."""

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from carddemo_batch.models.base import Base


class UserSecurity(Base):
    """
    User security entity representing the USRSEC VSAM file.
    
    Migrated from COBOL copybook CSUSR01Y.cpy.
    Record length: 80 bytes
    Primary key: SEC-USR-ID (8 characters)
    """
    __tablename__ = "usrsec"

    sec_usr_id: Mapped[str] = mapped_column(String(8), primary_key=True)
    sec_usr_fname: Mapped[str] = mapped_column(String(20), nullable=True)
    sec_usr_lname: Mapped[str] = mapped_column(String(20), nullable=True)
    sec_usr_pwd: Mapped[str] = mapped_column(String(8), nullable=True)
    sec_usr_type: Mapped[str] = mapped_column(String(1), nullable=True)

    USER_TYPE_ADMIN = "A"
    USER_TYPE_USER = "U"

    def is_admin(self) -> bool:
        """Check if user is an administrator."""
        return self.sec_usr_type == self.USER_TYPE_ADMIN

    def is_user(self) -> bool:
        """Check if user is a regular user."""
        return self.sec_usr_type == self.USER_TYPE_USER

    def __repr__(self) -> str:
        return f"UserSecurity(usr_id={self.sec_usr_id}, type={self.sec_usr_type})"
